### PR TITLE
CUDA 12.0 ARM conda support.

### PR DESCRIPTION
**Pull request by bdice**
This PR enables CUDA 12.0 ARM workflows for building and testing RAPIDS conda packages. This PR is currently targeting the 23.12 release.

### CODE REVIEW

**Comment by bdice** [comment]
Because ARM CI resources are limited, we will drop CUDA 11 ARM jobs for PRs before merging. We'll keep CUDA 11 ARM jobs in our nightly CI.

(discussed offline with @ajschmidt8)

File: .github/workflows/conda-cpp-tests.yaml
Comment (bdice) [inline .github/workflows/conda-cpp-tests.yaml:68]:
```suggestion
```

File: .github/workflows/conda-python-tests.yaml
Comment (bdice) [inline .github/workflows/conda-python-tests.yaml:71]:
```suggestion
```

**Review by bdice — COMMENTED** [review]
@ajschmidt8 @raydouglass This PR is ready for review. As previously discussed, I'm proposing to drop CUDA 11.8 ARM GPU tests from our PR matrix. It would remain in the nightly tests. Do we want to drop some ARM configuration from the nightly matrix?

**Comment by ajschmidt8** [comment]
> Do we want to drop some ARM configuration from the nightly matrix? I believe it currently includes ARM tests for 11.4, 11.8, and 12.0.

I'll defer to @raydouglass here. I wouldn't mind dropping some combinations from the matrix.

**Review by ajschmidt8 — APPROVED** [review]

**Comment by bdice** [comment]
I'll apply the changes to drop CUDA 11 ARM nightlies from PR builds, but I'll leave our nightly jobs in place for the moment. We can revisit that separately.

**Pull request merged**
